chain/network/src/peer_manager/tests.rs

```diff
@@ -586,3 +586,25 @@ async fn connection_spam_security_test() {
         c.handshake(&clock.clock()).await;
     }
 }
+
+#[tokio::test]
+async fn outbound_loop_connection() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+    let mut cfg = chain.make_config(rng);
+    cfg.node_key = pm.cfg.node_key.clone();
+
+    // Starting an outbound loop connection should be stopped without sending the handshake.
+    pm.start_outbound(chain.clone(), cfg).await.fail_handshake(&clock.clock()).await;
```

Comment:
Looking at this, I think there's one problem with `fail_handshake` testing API -- it doesn't check the *reason* why the connection got dropped. It'd be more robust if we somehow also verified here that this is due to LoopConnection error, and not something else.

Reply:
done, I had to make quite a few changes to fit these checks in.